runtime/resolvers/metricsview_sql.go

```diff
@@ -0,0 +1,257 @@
+package resolvers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	compilerv1 "github.com/rilldata/rill/runtime/compilers/rillv1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/exp/maps"
+)
+
+func init() {
+	runtime.RegisterAPIResolverInitializer("Metrics", newMetricsViewSQL)
+}
+
+// sqlIdenitifer is regex pattern to identify a SQL idenitifier. The idenitifier may be wrapped in double quotes.
+// Additionally if double quotes are present in idenitifer, it is escaped with additional double quotes.
+var sqlIdenitifer = `[a-zA-z_][a-zA-Z0-9_]*|"(?:[^"]|"")*"`
+
+var (
+	aggregateRegex = regexp.MustCompile(fmt.Sprintf(`(?i)AGGREGATE\((?:(%s)\.)?(%s)\)`, sqlIdenitifer, sqlIdenitifer))
+	fromMVRegex    = regexp.MustCompile(fmt.Sprintf(`(?i)FROM\s+(%s)`, sqlIdenitifer))
+)
+
+func newMetricsViewSQL(ctx context.Context, opts *runtime.APIResolverOptions) (runtime.APIResolver, error) {
+	sql := opts.API.Spec.ResolverProperties.Fields["sql"].GetStringValue()
+	if sql == "" {
+		return nil, errors.New("no sql query found for metrics sql resolver")
+	}
```

Comment:
1. I think there might be a risk of nil panics for this: `.ResolverProperties.Fields` (in case no properties are set, not sure how that serializes/deserializes).
2. Even though there's only `sql` now, I would consider adding a `metricsSQLProperties` struct and deserializing to that using `mapstructure`, similar to what we do for connector properties. It makes it easier to get an overview of available properties and to add new options.